### PR TITLE
Create foundations for REST API

### PR DESCRIPTION
**Pull request by lily-mayfield**
Do most the heavy work for implementing the OSP API. Creates a rudimentary
draft for the OSP API, which requires much review, includes lots of comments/notes/questions.

There's an issue with `flask_cache` where I am unable to get caching to working optimally in regard to querystrings (if in different order, creates new cache), which I believe can be resolved by "fixing" the cache key callback (perhaps by doing an `OrderedSet`?). Here are the lines in question:

  * The cache key maker: https://github.com/opensyllabus/osp-api/pull/2/files#diff-ab98d0d2e1ffd614f71caea6435e9e8fR4
  * Method using caching (with more details on issue!): https://github.com/opensyllabus/osp-api/pull/2/files#diff-495014a11d320df0b2b12c43f595615bR61

I would appreciate a look-over (not just pertaining to above issue) before I implement further.

I do not intend to merge this PR until I have completed the basic skeleton of this project, but am looking forward to amending this PR with commits in regard to review.

### CODE REVIEW

**Comment by lily-mayfield** [comment]
The caching issue was actually a problem with a tool I use to test the API :laughing:

**Comment by lily-mayfield** [comment]
It's worth noting that in the two new commits I've made I both solve the cache bug (hearing "rock the cache bug" in my head) and further developed the API spec which also included incorporating suggestions from the GENOSHA team.

**Comment by lily-mayfield** [comment]
Here's an issue: I'm using the outdated/unmaintained `flask-cache`, the new thing is `flask-caching`.

https://pythonhosted.org/Flask-Caching/

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Contributing to this project
+
+PEP8 is good. Try `pip install pep8` and `pep8 app.py`
+
+I use [Google-style
```
Comment (wearpants) [inline CONTRIBUTING.md]: We've been using standard Sphinx in some of the other OSP projects, but inconsistently.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+backend/REST API.
+
+For the REST API itself please take a look at
+[rest-api-spec.md](rest-api-spec.md).
+
+## Getting started
+
+`pip install -r requirements.txt`
+
+### Docker
```
Comment (wearpants) [inline README.md]: We've been using Vagrant for development.

File: CONTRIBUTING.md (hunk 1 above)
Comment (lily-mayfield) [inline CONTRIBUTING.md]: You can use Google-style docstrings with Sphinx using the Napoleon extension, it's great! It makes your docstrings more legible, IMO.

File: README.md (hunk 2 above)
Comment (lily-mayfield) [inline README.md]: Do you think I should switch this project over to Vagrant? I've never used that, but wouldn't mind learning something new. :)

**Pull request merged**